File: app/src/main/java/com/emsi/projetws/beans/Etudiant.java
package com.emsi.projetws.beans;

public class Etudiant {
    private int id;
    private String nom;
    private String prenom;
    private String ville;
    private String sexe;
    private String photo; // Base64 encoded string for now (optional)
    private String date_naissance;

    // Default constructor
    public Etudiant() {}

    // Parameterized constructor
    public Etudiant(int id, String nom, String prenom, String ville, String sexe, String photo, String date_naissance) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.ville = ville;
        this.sexe = sexe;
        this.photo = photo;
        this.date_naissance = date_naissance;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getSexe() {
        return sexe;
    }

    public void setSexe(String sexe) {
        this.sexe = sexe;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public String getDateNaissance() {
        return date_naissance;
    }

    public void setDateNaissance(String date_naissance) {
        this.date_naissance = date_naissance;
    }

    @Override
    public String toString() {
        return "Etudiant{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prenom='" + prenom + '\'' +
                ", ville='" + ville + '\'' +
                ", sexe='" + sexe + '\'' +
                ", date_naissance='" + date_naissance + '\'' +
                '}';
    }
}

File: projet/classes/Etudiant.php
<?php
class Etudiant {
    private $id;
    private $nom;
    private $prenom;
    private $ville;
    private $sexe;
    private $photo;
    private $date_naissance;
    
    function __construct($id, $nom, $prenom, $ville, $sexe, $photo = null, $date_naissance = null) {
        $this->id = $id;
        $this->nom = $nom;
        $this->prenom = $prenom;
        $this->ville = $ville;
        $this->sexe = $sexe;
        $this->photo = $photo;
        $this->date_naissance = $date_naissance;
    }
    
    function getId() {
        return $this->id;
    }
    
    function getNom() {
        return $this->nom;
    }
    
    function getPrenom() {
        return $this->prenom;
    }
    
    function getVille() {
        return $this->ville;
    }
    
    function getSexe() {
        return $this->sexe;
    }
    
    function getPhoto() {
        return $this->photo;
    }
    
    function getDateNaissance() {
        return $this->date_naissance;
    }
    
    function setId($id) {
        $this->id = $id;
    }
    
    function setNom($nom) {
        $this->nom = $nom;
    }
    
    function setPrenom($prenom) {
        $this->prenom = $prenom;
    }
    
    function setVille($ville) {
        $this->ville = $ville;
    }
    
    function setSexe($sexe) {
        $this->sexe = $sexe;
    }
    
    function setPhoto($photo) {
        $this->photo = $photo;
    }
    
    function setDateNaissance($date_naissance) {
        $this->date_naissance = $date_naissance;
    }
    
    public function __toString() {
        return $this->nom . " " . $this->prenom;
    }
}
